### DISCUSSION TITLE
Publishing Output of Yolo Model

### DISCUSSION THREAD

**Discussion by labaxisAI2** [AxisCommunications/axis-model-zoo#44] [Q&A]
Curious if its possible to take what the pose estimator does with publishing using flask (acap-computer-vision-sdk-sxamples) and using a similar approach with an individual _Yolov5_ model?
In axis-model-zoo, Axis generously provides ways to test inferencing of different public models using the `larod-client` on randomly generated images. If a model successfully pass the `model_performance_tester.py` script does this mean I can create a visualization of the models output through the camera?

Side question - Would it ever be possible to run individual models as an app on the Camera's GUI, like AOA or AVMD.

**Comment by Corallo** [comment]
Hi @labaxisAI2 

Thanks for your input,
I would break down the questions:
> Curious if its possible to take what the pose estimator does with publishing using flask (acap-computer-vision-sdk-sxamples) and using a similar approach with an individual _Yolov5_ model?

Yeah you can definitely swap the model in the pose estimator example, to replace it with yolo ([here](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/pose-estimator-with-flask/Dockerfile.model#L11)  and [here](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/pose-estimator-with-flask/config/env.aarch64.artpec8#L1)), and adjust the output of the model, ([here](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/blob/main/pose-estimator-with-flask/app/detector_with_flask.py#L124) to get yolo output instead of the keypoints).
We know it would be nice to have an end-to-end example with output visualization on yolo, but at the moment we don't have short terms plans to do it, other things have higher priority. We encourage you to try and ask for help in the [example repository](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/main) or in the [developer community](https://github.com/orgs/AxisCommunications/discussions) if you get stuck.

> In axis-model-zoo, Axis generously provides ways to test inferencing of different public models using the larod-client on randomly generated images. 

Yes, but larod-client can also be used with non-random-input (real images) see [here](https://axiscommunications.github.io/acap-documentation/docs/computer-vision-on-device/test-your-model.html) and [here](https://axiscommunications.github.io/acap-documentation/docs/api/src/api/larod/html/md__opt_builder-doc_larod_doc_larod-client.html) for more details.

> If a model successfully pass the model_performance_tester.py script does this mean I can create a visualization of the models output through the camera?

Not sure what you mean here, you could change that script to feed in input an image instead of random data, and visualize the results. however, we recommend this only for "experimenting" as it is not an efficient way.

> Side question - Would it ever be possible to run individual models as an app on the Camera's GUI, like AOA or AVMD.

Yes, you would have to make your own ACAP. Here you can find [documentation](https://axiscommunications.github.io/acap-documentation/) and examples [1](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples) [2](https://github.com/AxisCommunications/acap-native-sdk-examples)
Although it is possible, I am not sure we have an example on how to make a GUI in our repositories, the closest thing is probably the web-server example.
You might want to try asking this in the [developer community](https://github.com/orgs/AxisCommunications/discussions), this repo focuses more on model development and performance rather than ACAP development

**Reply by labaxisAI2** [reply]
Hi @Corallo ,
Thank you for the quick response and guidance, I will try this out!
Also, I was not concise with my question and included extra information, apologies for the confusion with those extra comments.
I have learned a great deal from all the clear guides you all provide, as well as the clear answers to the discussion questions.
Thank you again!
